File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationResponse(BaseModel):
    id: int
    type: str
    message: str
    is_read: bool
    actor_id: int | None = None
    file_id: int | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class NotificationListResponse(BaseModel):
    page: int
    limit: int
    total: int
    unread_count: int
    notifications: list[NotificationResponse]


class UnreadCountResponse(BaseModel):
    unread_count: int

File: frontend/src/types/notification.ts
export interface AppNotification {
  id: number;
  type: string;
  message: string;
  is_read: boolean;
  actor_id: number | null;
  file_id: number | null;
  created_at: string;
}

export interface NotificationList {
  page: number;
  limit: number;
  total: number;
  unread_count: number;
  notifications: AppNotification[];
}
